### ISSUE TITLE
Rule test with full config fails

### ISSUE THREAD

**Issue by jschaf** [eslint/eslint#6966]
I'm trying to test a full ESLint config with a bunch of small rules.  I created an empty rule and passed in the full config to RuleTester.setDefaultConfig.  But it doesn't seem to respect parserOptions and it runs into a type error.

**What version of ESLint are you using?**
v3.3.1
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**
Full setup at:
https://gist.github.com/jschaf/8ce46b7363510777f66b7a8eab730b43
**What did you do? Please include the actual source code causing the issue.**
Ran NPM test
**What did you expect to happen?**
All tests to pass:
**What actually happened? Please include the actual, raw output from ESLint.**
```sh
 npm test                                                                                              No es6 import export

> my-repo@0.0.1 test /tmp/rule-problem
> mocha test.js



  whitespace-characters
    valid
      1) var foo = 2;
      2) let foo = 2;
      3) const bar = 3;
    invalid
      4)        let foo = 2;


  0 passing (39ms)
  4 failing

  1) whitespace-characters valid var foo = 2;:
     TypeError: Error while loading rule 'whitespace-characters': Cannot convert undefined or null to object
      at Function.keys (native)
      at node_modules/eslint/lib/eslint.js:820:28
      at Array.forEach (native)
      at EventEmitter.module.exports.api.verify (node_modules/eslint/lib/eslint.js:792:16)
      at runRuleForItem (node_modules/eslint/lib/testers/rule-tester.js:324:38)
      at testValidTemplate (node_modules/eslint/lib/testers/rule-tester.js:357:28)
      at Context.<anonymous> (node_modules/eslint/lib/testers/rule-tester.js:455:25)

  2) whitespace-characters valid let foo = 2;:

      AssertionError: Should have no errors but had 1: [ { ruleId: null,
    fatal: true,
    severity: 2,
    source: 'let foo = 2;',
    message: 'Parsing error: Unexpected token foo',
    line: 1,
    column: 5 } ]
      + expected - actual

      -1
      +0

      at testValidTemplate (node_modules/eslint/lib/testers/rule-tester.js:360:20)
      at Context.<anonymous> (node_modules/eslint/lib/testers/rule-tester.js:455:25)

  3) whitespace-characters valid const bar = 3;:

      AssertionError: Should have no errors but had 1: [ { ruleId: null,
    fatal: true,
    severity: 2,
    source: 'const bar = 3;',
    message: 'Parsing error: The keyword \'const\' is reserved',
    line: 1,
    column: 1 } ]
      + expected - actual

      -1
      +0

      at testValidTemplate (node_modules/eslint/lib/testers/rule-tester.js:360:20)
      at Context.<anonymous> (node_modules/eslint/lib/testers/rule-tester.js:455:25)

  4) whitespace-characters invalid      let foo = 2;:

      AssertionError: A fatal parsing error occurred: Parsing error: Unexpected token foo
      + expected - actual

      -false
      +true

      at testInvalidTemplate (node_modules/eslint/lib/testers/rule-tester.js:392:28)
      at Context.<anonymous> (node_modules/eslint/lib/testers/rule-tester.js:463:25)



npm ERR! Test failed.  See above for more details.

```

**Comment by vitorbal** [comment]
I think the problem is that the `create` function needs to always return an object.
In your `test.js` file, If you change this line:

```javascript
create(context) {},
```

To this:

```javascript
create(context) { return {}; }
```

Then you shouldn't see an error anymore. Could you try that?

**Comment by jschaf** [comment]
Yes, that fixes the error for the first test case.  I still have the same errors for  2-4, namely ESLint isn't using the parserOptions.ecmaVersion for the test cases.

**Comment by nzakas** [comment]
Your ordering is wrong. You have:

```js
const RuleTester = require('eslint').RuleTester;
const ruleTester = new RuleTester();
RuleTester.setDefaultConfig(myConfig);
```

It needs to be:

```js
const RuleTester = require('eslint').RuleTester;
RuleTester.setDefaultConfig(myConfig);
const ruleTester = new RuleTester();
```

You were creating your instance first, then setting the default config for all instances, so it was too late.

**Comment by jschaf** [comment]
Ah, that's it!  Thank you for the help.
